sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -1662,33 +1665,66 @@ private synchronized void postponeRelease(ReleaseDetails releaseDetails) {
-                resumeDownload();
+            if (!InstallerUtils.isUnknownSourcesEnabled(mContext)) {
+                showUnknownSourcesDialog();
+                return;
+            }
+            requestPermissionsForDownload();
+            AppCenterLog.debug(LOG_TAG, "Schedule download...");
+            resumeDownload();
 
-                /* Refresh mandatory dialog progress or do nothing otherwise. */
-                showDownloadProgress();
+            /* Refresh mandatory dialog progress or do nothing otherwise. */
+            showDownloadProgress();
 
-                /*
-                 * If we restored a cached dialog, we also started a new check release call.
-                 * We might have time to click on download before the call completes (easy to
-                 * reproduce with network down).
-                 * In that case the download will start and we'll see a new update dialog if we
-                 * don't cancel the call.
-                 */
-                if (mCheckReleaseApiCall != null) {
-                    mCheckReleaseApiCall.cancel();
-                }
-            } else {
-                showUnknownSourcesDialog();
+            /*
+             * If we restored a cached dialog, we also started a new check release call.
+             * We might have time to click on download before the call completes (easy to
+             * reproduce with network down).
+             * In that case the download will start and we'll see a new update dialog if we
+             * don't cancel the call.
+             */
+            if (mCheckReleaseApiCall != null) {
+                mCheckReleaseApiCall.cancel();
             }
         } else {
             showDisabledToast();
         }
     }
 
+    private void requestPermissionsForDownload() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) {
+            AppCenterLog.debug(LOG_TAG, "There is no need to request permissions in runtime on Android earlier than 6.0.");
+            return;
+        }
+        if (PermissionUtils.permissionsAreGranted(mContext, Manifest.permission.POST_NOTIFICATIONS)) {
+            AppCenterLog.debug(LOG_TAG, "Post notification permission already granted.");
+            return;
+        }
+        AppCenterFuture<PermissionRequestActivity.Result> confirmFuture = PermissionUtils.requestPermissions(mContext, Manifest.permission.POST_NOTIFICATIONS);
+        if (confirmFuture == null) {
+            AppCenterLog.error(LOG_TAG, "Future to get the result of a permission request is null.");
+            return;
+        }
+        confirmFuture.thenAccept(new AppCenterConsumer<PermissionRequestActivity.Result>() {
+
+            @Override
+            public void accept(PermissionRequestActivity.Result result) {
+                if (result == null) {
+                    AppCenterLog.warn(LOG_TAG, "Failed to get result of attempt to get permissions.");
```

Comment:
This case should not be possible. I'd add @NonNull annotation here and in Activity class